static/app/main.tsx

```diff
@@ -7,15 +7,18 @@ import ConfigStore from 'sentry/stores/configStore';
 import {PersistedStoreProvider} from 'sentry/stores/persistedStore';
 import {RouteContext} from 'sentry/views/routeContext';
 
+import RouteAnalyticsContextProvider from './views/routeAnalyticsContextProvider';
 /**
  * Renders our compatability RouteContext.Provider. This will go away with
  * react-router v6.
  */
 function renderRouter(props: any) {
   return (
-    <RouteContext.Provider value={props}>
-      <RouterContext {...props} />
-    </RouteContext.Provider>
+    <RouteAnalyticsContextProvider {...props}>
```

Comment:
Note that the provider has to be above the `RouterContext` rather than using a hook inside of `makeLazyloadComponent` because we might have multiple route matches. To get around this we use a context provider that will be driven from a hook in Getsentry.